### ISSUE TITLE
InputNumber editor in Datatable doesn't accept only numbers as input values anymore

### ISSUE THREAD

**Issue by ursfan** [primefaces/primereact#2631]
### There is no guarantee in receiving an immediate response in GitHub Issue Tracker, If you'd like to secure our response, you may consider *PrimeReact PRO Support* where support is provided within 4 business hours

**I'm submitting a ...**  (check one with "x")
```
[x] bug report
[ ] feature request
[ ] support request => Please do not submit support request here, instead see https://forum.primefaces.org/viewforum.php?f=57
```

**Codesandbox Case (Bug Reports)**
https://codesandbox.io/s/92k7m

**Current behavior**
Using an InputNumber component as an editor of a table cell doesn't work correctly anymore. Starting with version 7.2. it is no longer possible that input of characters other than numbers are corrected by the input field. Instead the characters for example an 'a' or any other still remains in the field and only is deleted when a number was put in or the editor is closed.

**Expected behavior**
As in the versions before it shouldn't be possible to input any other characters than values. The field should delete wrong input characters while editing a field.

**Minimal reproduction of the problem with instructions**
I simply link the edit example of the Datatable from the showcase for version 7.2 to this issue. When one tries to input any characters to the fields quantity or price of a table cell in the first example or any other it can be seen easily.

* **React version:**
17.0.2

* **PrimeReact version:**
7.2.0

* **Browser:** [all | Chrome XX | Firefox XX | IE XX | Safari XX | Mobile Chrome XX | Android X.X Web Browser | iOS XX Safari | iOS XX UIWebView | iOS XX WKWebView ]
all

**Comment by melloware** [comment]
Isn't this a duplicate of #2532 ???

PR: https://github.com/primefaces/primereact/pull/2533

**Comment by ursfan** [comment]
I don't think its an duplicate. In my opinion the problem described here is a problem which only appears in the context of a Datatable. It seems the validateValue function of the InputNumber field is called but the result isn't shown in the editor field of the table. But when I debug that issue, the correct result is shown. So it seems to me that it depends on the waiting time, means there should be a waiting time for the return value of InputNumber before the table shows the input to work.

**Comment by ursfan** [comment]
I wondered why it works on the official example of PrimeReact 7.2.0 for InputNumber. But at a closer look it shows that the linked example is with version 6.6.0 of PrimeReact instead of 7.2.0. So it seems it wasn't yet updated.

**Comment by melloware** [comment]
@ursfan can you check the other issue also https://github.com/primefaces/primereact/issues/2633 I think they are both related to editing Input Number's in DataTable.

**Comment by ursfan** [comment]
I compared the esm module file for InputNumber of version 6.6.0 and 7.2.0 there is one difference in the function "onInputKeyPress" which seems to be responsible for the change of behavior of this component. In the version 6.6.0 the function event.preventDefault() is called without any condition and in the version 7.2.0 it is only called when there is a valid input. When I change that  in version 7.2.0 back as in 6.6.0 it works for me. So I patch my version until a new PrimeReact Version gives a patch for that.

**Comment by melloware** [comment]
@ursfan looks like it was this fix included in 7.0.0 that is the change: https://github.com/primefaces/primereact/issues/2344

**Comment by ursfan** [comment]
So when the fix for #2344 causes that error there was a reason to change the function "onInputKeyPress" for that. So when the solution of IHPNULL preserves the solution which was intended by the fix, it should be preferred.
